docs/running-rasa-with-docker.rst

```diff
@@ -0,0 +1,435 @@
+    :ref:`cli-usage`.
+
+
+Using Docker Compose to Run Multiple Services
+---------------------------------------------
+
+To run Rasa together with other services, such as a server for custom actions, it is
+recommend to use `Docker Compose <https://docs.docker.com/compose/>`_.
+Docker Compose provides an easy way to run multiple containers together without
+having to run multiple commands.
+
+Start by creating a file called ``docker-compose.yml``:
+
+.. code-block:: bash
+
+  touch docker-compose.yml
+
+Add the following content to the file:
+
+.. code-block:: yaml
+
+  version: '3.0'
+  services:
+    rasa:
+      image: rasa/rasa:latest-full
+      ports:
+        - 5005:5005
+      volumes:
+        - ./:/app
+      command:
+        - run
+
+
+The file starts with the version of the Docker Compose specification that you
+want to use.
+Each container is declared as a ``service`` within the docker compose file.
+The first service is the ``rasa`` service.
+
+
+The command is similar to the ``docker run`` command in :ref:`running_docker_container`.
+The ``ports`` part defines a port mapping between the container and your host
+system. In this case it makes ``5005`` of the ``rasa`` service available on
+port ``5005`` of your host.
+This is the port of the :ref:`rest_channels` interface of Rasa.
+
+.. note::
+
+    Since Docker Compose starts a set of Docker containers, it is no longer
+    possible to directly connect to one single container after executing the
+    ``run`` command.
+
+To run the services configured in your ``docker-compose.yml`` execute:
+
+.. code-block:: bash
+
+    docker-compose up
+
+
+Adding Custom Actions
+---------------------
+
+To create more sophisticated assistants you will want to use :ref:`custom-actions`.
+Continuing the example from above you might want to add an action which tells
+the user a joke to cheer the user up.
+
+Creating a Custom Action
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+Start with creating the custom actions in a directory ``actions``:
+
+.. code-block:: bash
+
+  mkdir actions
+  # Rasa Core SDK expects a python module.
+  # Therefore, make sure that you have this file in the directory.
+  touch actions/__init__.py
+  touch actions/actions.py
+
+Then build a custom action using the Rasa Core SDK, e.g.:
+
+.. code-block:: python
+
+  import requests
+  import json
+  from rasa_core_sdk import Action
+
+
+  class ActionJoke(Action):
+    def name(self):
+      return "action_joke"
+
+    def run(self, dispatcher, tracker, domain):
+      request = requests.get('http://api.icndb.com/jokes/random').json()  # make an api call
+      joke = request['value']['joke']  # extract a joke from returned json response
+      dispatcher.utter_message(joke)  # send the message back to the user
+      return []
+
+Next add the custom action in your stories and your domain file.
+Continuing with the example bot from ``rasa init``, replace ``utter_cheer_up`` in
```

Comment:
Changed to reflect the first part of the guide, need to check to make sure this paragraph still makes sense